prestoadmin/util/presto_config.py

```diff
@@ -45,28 +45,38 @@ class PrestoConfig:
         CLIENT_KEYSTORE_PASSWORD_KEY: None,
     }
 
-    def __init__(self, config_properties):
+    def __init__(self, config_properties, config_host, config_path):
+        self.config_path = config_path
+        self.config_host = config_host
         if not config_properties:
             self.config_properties = self.default_config
         else:
             self.config_properties = config_properties
 
     @staticmethod
     def coordinator_config():
+        config_path = os.path.join(REMOTE_CONF_DIR, CONFIG_PROPERTIES)
+        config_host = env.roledefs['coordinator'][0]
         try:
             data = StringIO()
-            with settings(host_string='%s@%s' % (env.user, env.roledefs['coordinator'][0])):
-                get(os.path.join(REMOTE_CONF_DIR, CONFIG_PROPERTIES), data)
+            with settings(host_string='%s@%s' % (env.user, config_host)):
+                get(config_path, data)
 
             data.seek(0)
             presto_config_dict = get_conf_from_properties_data(data)
         except:
             _LOGGER.info('Could not find Presto config.')
             presto_config_dict = None
-        return PrestoConfig(presto_config_dict)
+        return PrestoConfig(presto_config_dict, config_path, config_host)
 
     def _lookup(self, key):
-        return self.config_properties.get(key, self.default_config[key])
+        result = self.config_properties.get(key, self.default_config[key])
+        if not result:
+            raise Exception(
+                    "Key %s is not configured in coordinator configuration"
+                    "%s on host %s and has no default" %
```

Comment:
Maybe we should remove the default for CLIENT_KEYSTORE_PATH_KEY/CLIENT_KEYSTORE_PASSWORD_KEY/?